### ISSUE TITLE
Vulkan Mobile: LightmapGI sampling is not disabled on lights with Static bake mode, causing double lighting to appear

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#85145]
### Godot version

4.2.rc1

### System information

Fedora 38, GeForce RTX 4090 (NVIDIA 535.129.03)

### Issue description

LightmapGI sampling is not disabled on lightmapped meshes with Static bake mode, causing double lighting to appear:

Forward+ | Mobile
-|-|-
![Screenshot_20231120_175447](https://github.com/godotengine/godot/assets/180032/fc4c2c79-fa9e-4d42-ba2a-d1b6ebf71ada) | ![Screenshot_20231120_175831](https://github.com/godotengine/godot/assets/180032/3d62d218-6d82-4a64-b04e-30deecfc807c)

### Steps to reproduce

- Create meshes with Static bake mode that have UV2 (you can use **Add UV2** in PrimitiveMesh for this).
- Add some lights, enable shadows on them and set their bake mode to **Static**.
- Bake lightmaps and look at the scene's appearance.
- Switch to the Mobile rendering method and compare visuals.

### Minimal reproduction project

[test_lightmapgi_mobile.zip](https://github.com/godotengine/godot/files/13417782/test_lightmapgi_mobile.zip)

**Comment by clayjohn** [comment]
To clarify, the problem here is that the light is still drawn dynamically despite using the static bake mode right? If a light is statically baked, my expectation is that only the lightmap will be used. I wouldn't expect the lightmap to be disabled.

I noticed while working on something else that the dynamic light is added to the lightmap even when the light is statically baked

**Comment by Calinou** [comment]
> To clarify, the problem here is that the light is still drawn dynamically despite using the static bake mode right? If a light is statically baked, my expectation is that only the lightmap will be used. I wouldn't expect the lightmap to be disabled.

Yes, I expect it to render like in Forward+ and Compatibility where the real-time light is skipped on the lightmapped surface (since it's already been baked).

**Comment by clayjohn** [comment]
To be clear, in the Forward+ backend, the dynamic light is still added on lightmapped surfaces. So this issue must exist in both backends. 

Forward+ also has an unitialized memory error because the bake mode is only passed to the shader if the light is using shadows.

**Comment by Kaleb-Reid** [comment]
@clayjohn Was this fixed by #96771?

**Comment by clayjohn** [comment]
Not sure. I thought it was. But I could never repro the exact issue here. I fixed the only issue that I could repro. 

@Calinou will have to confirm if this is fixed

**Comment by Calinou** [comment]
I can confirm this is fixed as of 4.4.1.stable (still works in 4.6.dev too).
